[
  {
    "key": 289003857, 
    "kind": "ERROR", 
    "hash": 651579144, 
    "severity": "HIGH", 
    "bug_trace": [
      {
        "node_tags": [
          {
            "tag": "kind", 
            "value": "procedure_start"
          }, 
          {
            "tag": "name", 
            "value": "void CheckSqlRawqueryRuleTest.test()"
          }, 
          {
            "tag": "name_id", 
            "value": "com.qihoo.ruletestcases.CheckSqlRawqueryRuleTest.test():void.09c655043869554424bb60b2005c8e18"
          }
        ], 
        "line_number": 16, 
        "filename": "app/src/main/java/com/qihoo/ruletestcases/CheckSqlRawqueryRuleTest.java", 
        "description": "start of procedure test()", 
        "level": 0
      }, 
      {
        "node_tags": [], 
        "line_number": 17, 
        "filename": "app/src/main/java/com/qihoo/ruletestcases/CheckSqlRawqueryRuleTest.java", 
        "description": "", 
        "level": 0
      }
    ], 
    "bug_class": "PROVER", 
    "bug_type": "RESOURCE_LEAK", 
    "file": "app/src/main/java/com/qihoo/ruletestcases/CheckSqlRawqueryRuleTest.java", 
    "qualifier_tags": [
      {
        "tag": "line", 
        "value": "17"
      }, 
      {
        "tag": "call_procedure", 
        "value": "rawQuery(...)"
      }, 
      {
        "tag": "call_line", 
        "value": "17"
      }
    ], 
    "line": 17, 
    "procedure_id": "com.qihoo.ruletestcases.CheckSqlRawqueryRuleTest.test():void.09c655043869554424bb60b2005c8e18", 
    "procedure": "void CheckSqlRawqueryRuleTest.test()", 
    "qualifier": " resource of type android.database.sqlite.SQLiteCursor acquired by call to rawQuery(...) at line 17 is not released after line 17"
  }, 
  {
    "key": 174956197, 
    "kind": "ERROR", 
    "hash": 25238872, 
    "severity": "HIGH", 
    "bug_trace": [
      {
        "node_tags": [
          {
            "tag": "kind", 
            "value": "procedure_start"
          }, 
          {
            "tag": "name", 
            "value": "void NullPointerTest.test1()"
          }, 
          {
            "tag": "name_id", 
            "value": "com.qihoo.ruletestcases.NullPointerTest.test1():void.dca82ea974b8d8a60a264206f8d1c77d"
          }
        ], 
        "line_number": 16, 
        "filename": "app/src/main/java/com/qihoo/ruletestcases/NullPointerTest.java", 
        "description": "start of procedure test1()", 
        "level": 0
      }, 
      {
        "node_tags": [], 
        "line_number": 17, 
        "filename": "app/src/main/java/com/qihoo/ruletestcases/NullPointerTest.java", 
        "description": "", 
        "level": 0
      }, 
      {
        "node_tags": [], 
        "line_number": 18, 
        "filename": "app/src/main/java/com/qihoo/ruletestcases/NullPointerTest.java", 
        "description": "", 
        "level": 0
      }
    ], 
    "bug_class": "PROVER", 
    "bug_type": "NULL_DEREFERENCE", 
    "file": "app/src/main/java/com/qihoo/ruletestcases/NullPointerTest.java", 
    "qualifier_tags": [
      {
        "tag": "bucket", 
        "value": "B1"
      }, 
      {
        "tag": "line", 
        "value": "18"
      }, 
      {
        "tag": "assigned_line", 
        "value": "17"
      }, 
      {
        "tag": "value", 
        "value": "list"
      }
    ], 
    "line": 18, 
    "procedure_id": "com.qihoo.ruletestcases.NullPointerTest.test1():void.dca82ea974b8d8a60a264206f8d1c77d", 
    "procedure": "void NullPointerTest.test1()", 
    "qualifier": "object list last assigned on line 17 could be null and is dereferenced at line 18"
  }
]